docs/mobile-embed-frame.md

```diff
@@ -0,0 +1,38 @@
+---
+title: Mobile Embed
+layout: base
+load_script: false
+---
+
+<div id="preview"></div>
+<script>
+window.addEventListener('message', (event) => {
+  if (event.origin !== window.location.origin) {
+    return;
+  }
+
+  let markup;
+  try {
+    markup = JSON.parse(event.data).markup;
+  } catch {
+    return;
+  }
+
+  const preview = document.getElementById('preview');
+  preview.innerHTML = markup;
+  preview.querySelectorAll('script').forEach((script) => {
+    const replacement = document.createElement('script');
+    replacement.innerHTML = script.innerHTML;
+    script.parentNode.replaceChild(replacement, script);
+  });
```

Comment:
I know your [original comment mentioned this PR should raise some security eyebrows](https://github.com/18F/identity-design-system/pull/418#:~:text=For%20the%20security%2Dminded%2C%20this%20should%20probably%20raise%20some%20hairs)... just wanted to confirm yes eyebrows raised

Reply:
Yeah, on this code in particular, it's strange that [browsers don't evaluate dynamically inserted script tags](https://stackoverflow.com/questions/1197575/can-scripts-be-inserted-with-innerhtml), while seemingly having no issue with inserting something equally exploitable like `<img onload="somejavascript">`.

An alternative approach could be to have just a standalone HTML page that already has the component markup in it, though that might end up with a lot more duplication, especially if we wanted to use it for more components.

I might time-box and see if I can put something together for that and what it ends up looking like. I tried in an earlier iteration and had some trouble, but I think I was fighting more with Jekyll than the approach itself.

Reply:
For what it's worth, [I've implemented something like this in the past](https://github.com/Automattic/wp-calypso/pull/4619) and had previously spent a lot of time verifying the concept as safe, even with untrusted content in a secure context.

Although now that I write this, I probably need to harden this a bit more for the implementation to be _truly safe_ (specifically [`iframe` `sandbox` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#sandbox)). I'll see where I end up with my timeboxed experimentation whether this is an approach we go with.

Reply:
Updated in ace14d0 to avoid the `postMessage` communication and dynamic script insertion, and instead using a simple static page for the frame.